src/lib/service/learnerRecordAPI/moduleRecord/models/moduleRecord.ts

```diff
@@ -0,0 +1,69 @@
+import {Record, RecordState} from '../../models/record'
+
+export enum ModuleRecordResult {
+	Failed = 'FAILED',
+	Passed = 'PASSED',
+}
+
+export enum BookingStatus {
+	REQUESTED = 'REQUESTED',
+	CONFIRMED = 'CONFIRMED',
+	CANCELLED = 'CANCELLED',
+}
+
+export class ModuleRecord extends Record {
+	id: number
+	completionDate?: Date
+	eventId?: string
+	eventDate?: Date
+	moduleId: string
+	moduleTitle: string
+	moduleType: string
+	optional: boolean
```

Comment:
Don't know if it's possible but maybe `isOptional` is clearer since this is a boolean?

Reply:
I agree on this, but for now I'd prefer this to match the learner record interface (which in turn matches the DB) - until we clean up our database column naming conventions